index.js

```diff
@@ -19,100 +13,119 @@ const npar = spec => {
+        scope,
+        spec,
+        ignoreBody: true
+      })
+        .then(() => resolve(true))
+        .catch(reject)
+    } catch (err) {
+      return reject(err)
     }
-    const uri = `/-/team/${eu(scope)}/${eu(team)}/package`
-    return npmFetch(uri, opts.concat({
-      method: 'PUT',
-      body: { package: spec.name, permissions },
-      scope,
-      spec,
-      ignoreBody: true
-    }))
-  }).then(() => true)
+  })
 }
 
-cmd.revoke = (spec, entity, opts) => {
-  opts = AccessConfig(opts)
-  return pwrap(opts, () => {
-    spec = npar(spec)
-    const { scope, team } = splitEntity(entity)
-    validate('OSSO', [spec, scope, team, opts])
-    const uri = `/-/team/${eu(scope)}/${eu(team)}/package`
-    return npmFetch(uri, opts.concat({
-      method: 'DELETE',
-      body: { package: spec.name },
-      scope,
-      spec,
-      ignoreBody: true
-    }))
-  }).then(() => true)
+cmd.revoke = (spec, entity, opts = {}) => {
+  return new Promise((resolve, reject) => {
+    try {
+      spec = npar(spec)
+      const { scope, team } = splitEntity(entity)
+      validate('OSSO', [spec, scope, team, opts])
+      const uri = `/-/team/${eu(scope)}/${eu(team)}/package`
+      return npmFetch(uri, {
+        ...opts,
+        method: 'DELETE',
+        body: { package: spec.name },
+        scope,
+        spec,
+        ignoreBody: true
+      })
+        .then(() => resolve(true))
+        .catch(reject)
+    } catch (err) {
+      return reject(err)
+    }
+  })
 }
 
 cmd.lsPackages = (entity, opts) => {
-  opts = AccessConfig(opts)
-  return pwrap(opts, () => {
-    return getStream.array(
-      cmd.lsPackages.stream(entity, opts)
-    ).then(data => data.reduce((acc, [key, val]) => {
-      if (!acc) {
-        acc = {}
-      }
-      acc[key] = val
-      return acc
-    }, null))
+  return new Promise((resolve, reject) => {
+    return cmd.lsPackages.stream(entity, opts)
+      .collect()
+      .then(data => {
+        const packageList = data.reduce((acc, [key, val]) => {
+          if (!acc) {
+            acc = {}
+          }
+          acc[key] = val
+          return acc
+        }, null)
+        return resolve(packageList)
+      })
+      .catch(reject)
   })
 }
 
-cmd.lsPackages.stream = (entity, opts) => {
+cmd.lsPackages.stream = (entity, opts = {}) => {
   validate('SO|SZ', [entity, opts])
-  opts = AccessConfig(opts)
   const { scope, team } = splitEntity(entity)
   let uri
   if (team) {
     uri = `/-/team/${eu(scope)}/${eu(team)}/package`
   } else {
     uri = `/-/org/${eu(scope)}/package`
   }
-  opts = opts.concat({
-    query: { format: 'cli' },
-    mapJson (value, [key]) {
-      if (value === 'read') {
-        return [key, 'read-only']
-      } else if (value === 'write') {
-        return [key, 'read-write']
-      } else {
-        return [key, value]
-      }
-    }
-  })
-  const ret = new PassThrough({ objectMode: true })
+  opts.query = { format: 'cli' }
+  opts.mapJSON = mapJSON
```

Comment:
I think we want to not mutate the opts here, because it hasn't been cloned.  Safer to do:

```js
opts = {
  ...opts,
  query: { format: 'cli' },
  mapJSON
}
```